src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1334,15 +1348,19 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
       const gasUsedPreviousStep = totalGasUsedAfterThisStep - gas;
       gas += gasUsedPreviousStep;
 
-      const memory: ITraceData[] = [];
-      if (options.disableMemory !== true) {
+      let memory: ITraceData[];
+      if (options.disableMemory === true) {
+        memory = [];
+      } else {
         // We get the memory as one large array.
         // Let's cut it up into 32 byte chunks as required by the spec.
+        const limit = Number(event.memoryWordCount);
+        memory = Array(limit);
         let index = 0;
-        while (index < event.memory.length) {
-          const slice = event.memory.slice(index, index + 32);
-          memory.push(TraceData.from(Buffer.from(slice)));
-          index += 32;
+        while (index < limit) {
```

Comment:
Ethereum JS changed the way they slice and dice memory, and the `event.memory` Buffer here might be waaaaayyy too big (like 80x+ times too big), causing us to copy the whole Buffer, instead of just the actual current memory range (which can be computed by using `memoryWordCount`).

Reply:
Normally, yes, however, EJS has already fixed this bug upstream - they just haven't released the fix yet.

Anyway, I _do_ test this change now in the "should trace a successful transaction without changing state" test. It's just not a test specifically for this workaround, but more of a "does tracing work as expected" test.